java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java

```diff
@@ -148,4 +150,28 @@ public static boolean isOne(ExpressionTree tree) {
   public static boolean isNegOne(ExpressionTree tree) {
     return tree.is(Kind.UNARY_MINUS) && isOne(((UnaryExpressionTree) tree).expression());
   }
+  
+  public static String getAsStringValue(LiteralTree tree) {
+    if (!tree.is(Kind.TEXT_BLOCK)) {
+      return trimQuotes(tree.value());
+    }
+    String[] lines = trimQuotes(tree.value()).split("\r?\n");
+    int indent = Arrays.stream(lines).skip(1).mapToInt(LiteralUtils::getIndentation)
```

Comment:
I missed this before, but empty lines (or lines containing only white space) are ignored for the purposes of determining indentation. So if all non-empty lines are indented by 4 spaces and all empty lines contain no spaces (which will be common since many editors/IDEs remove trailing spaces), the indentation should be 4, not 0.